packages/theme-addresses/addressForm.ts

```diff
@@ -0,0 +1,274 @@
+
+  if (COUNTRIES) {
+    init(rootEl, formElements, options);
+  } else {
+    loadCountries(locale).then(function(countries) {
+      COUNTRIES = countries;
+      init(rootEl, formElements, options);
+    });
+  }
+}
+
+/**
+ * Runs when countries have been loaded
+ */
+function init(rootEl: HTMLElement, formElements: FormElements, options: AddressFormOptions) {
+  populateCountries(formElements);
+  var selectedCountry = formElements.country.input
+    ? formElements.country.input.value
+    : null;
+  handleCountryChange(rootEl, formElements, selectedCountry);
+  if (options.loaded) {
+    options.loaded();
+  }
+}
+
+/**
+ * Handles when a country change: set labels, reorder fields, populate zones
+ */
+function handleCountryChange(rootEl: HTMLElement, formElements: FormElements, countryCode: string) {
+  var country = getCountry(countryCode);
+
+  setEventListeners(rootEl, formElements, country);
+  setLabels(formElements, country);
+  reorderFields(rootEl, formElements, country);
+  populateZones(formElements, country);
+}
+
+/**
+ * Sets up event listener for country change
+ */
+function setEventListeners(rootEl: HTMLElement, formElements: FormElements, country: Country) {
+  formElements.country.input.addEventListener('change', function(event) {
+    handleCountryChange(rootEl, formElements, (event.target as HTMLSelectElement).value);
+  });
+}
+
+/**
+ * Reorder fields in the DOM and add data-attribute to fields given a country
+ */
+function reorderFields(rootEl: HTMLElement, formElements: FormElements, country: Country) {
+  var formFormat = country.formatting.edit;
+  var rowIndex = 0;
+
+  var countryWrapper = formElements.country.wrapper;
+  var afterCountry = false;
+  getOrderedField(formFormat).forEach(function(row) {
+    row.forEach(function(line) {
+      formElements[line].wrapper.dataset.lineCount = row.length;
+      if (!formElements[line].wrapper) {
+        return;
+      }
+      if (line === 'country') {
+        afterCountry = true;
+        return;
+      }
+
+      if (afterCountry) {
+        rootEl.append(formElements[line].wrapper);
+      } else {
+        rootEl.insertBefore(formElements[line].wrapper, countryWrapper);
+      }
+    });
+  });
+}
+
+/**
+ * Update labels for a given country
+ */
+function setLabels(formElements: FormElements, country: Country) {
+  Object.keys(formElements).forEach(function(formElementName) {
+    formElements[formElementName].labels.forEach(function(label) {
+      label.textContent = country.labels[formElementName];
+    });
+  });
+}
+
+/**
+ * Add right countries in the dropdown for a given country
+ */
+function populateCountries(formElements: FormElements) {
+  var countrySelect = formElements.country.input;
+
+  COUNTRIES.forEach(function(country) {
+    var optionElement = document.createElement('option');
+    optionElement.value = country.code;
+    optionElement.textContent = country.name;
+    countrySelect.appendChild(optionElement);
```

Comment:
You should create a duplicate of `countrySelect` in memory, append the new options to it, and then replace the version of `countrySelect` in the DOM with your updated one. This will significantly reduce the number of DOM operations you are performing.